src/MongoDB.Analyzer/Core/TypesProcessor.cs

```diff
@@ -283,4 +317,67 @@ TypeKind.Enum or
 
     private string GetFullName(ITypeSymbol typeSymbol) =>
         typeSymbol.ToDisplayString();
+
+    private AttributeListSyntax GenerateBsonAttributeList(ISymbol symbol)
+    {
+        var generatedBsonAttributes = new List<AttributeSyntax>();
+        var bsonAttributes = symbol.GetAttributes().Where(attribute => attribute.AttributeClass.IsSupportedBsonAttribute());
+
+        foreach (var bsonAttribute in bsonAttributes)
+        {
+            var bsonAttributeArgumentList = new List<AttributeArgumentSyntax>();
+
+            bsonAttributeArgumentList.AddRange(bsonAttribute.ConstructorArguments.Select(bsonAttributeArgument => GenerateBsonAttributeArgument(bsonAttributeArgument, false, null)));
+            bsonAttributeArgumentList.AddRange(bsonAttribute.NamedArguments.Select(bsonAttributeArgument => GenerateBsonAttributeArgument(bsonAttributeArgument.Value, true, bsonAttributeArgument.Key)));
+
+            var bsonAttributeSyntaxNode = SyntaxFactoryUtilities.GetAttribute(bsonAttribute.AttributeClass.Name, bsonAttributeArgumentList);
+            generatedBsonAttributes.Add(bsonAttributeSyntaxNode);
+        }
+
+        var bsonAttributeList = SyntaxFactory.AttributeList(SyntaxFactory.SeparatedList<AttributeSyntax>(generatedBsonAttributes));
```

Comment:
small optimization, we can skip creating `bsonAttributeList` and `generatedBsonAttributes` if not needed.

Reply:
Done